src/java/picard/analysis/CollectMultipleMetrics.java

```diff
@@ -184,7 +217,10 @@ public int doWork() {
 
         final List<SinglePassSamProgram> programs = new ArrayList<SinglePassSamProgram>();
         for (final ProgramInterface program : new HashSet<ProgramInterface>(programsToRun)) {
-            final SinglePassSamProgram instance = program.makeInstance(OUTPUT , INPUT, REFERENCE_SEQUENCE);
+            if (program.needsReferenceSequence() && REFERENCE_SEQUENCE==null) {
+                throw new PicardException("A program you are trying to run needs a Reference Sequence, please set REFERENCE_SEQUENCE in the command line");
```

Comment:
Yes it does, I figured that out finally and added a new commit where it does that a little while ago. Thanks Yossi! 



> On Jun 18, 2015, at 4:52 PM, Yossi Farjoun <notifications@github.com> wrote:
> 
> In src/java/picard/analysis/CollectMultipleMetrics.java:
> 
> > @@ -184,7 +217,10 @@ public int doWork() {
> >  
> >          final List<SinglePassSamProgram> programs = new ArrayList<SinglePassSamProgram>();
> >          for (final ProgramInterface program : new HashSet<ProgramInterface>(programsToRun)) {
> > -            final SinglePassSamProgram instance = program.makeInstance(OUTPUT , INPUT, REFERENCE_SEQUENCE);
> > +            if (program.needsReferenceSequence() && REFERENCE_SEQUENCE==null) {
> > +                throw new PicardException("A program you are trying to run needs a Reference Sequence, please set REFERENCE_SEQUENCE in the command line");
> won't program.toString() do it?
> …
> —
> Reply to this email directly or view it on GitHub.
>